_posts/2016-08-15-arbitrary-cpu-ram.md

```diff
@@ -0,0 +1,91 @@
+---
+author: <a href="https://github.com/ivanlomba">Iván Lomba</a>
+comments: true
+date: 2016-08-15 07:00:00+00:00
+layout: post
+slug: arbitrary-cpu-ram
+title: Arbitrary CPU and RAM supported in the ComputeService
+---
+
+As part of a [Google Summer of Code](https://developers.google.com/open-source/gsoc/) project has been added a feature to allow users to set manually specific values of CPU and RAM.
+<!--more-->
+
+The previous Compute Service abstraction assumed that all providers had hardware profiles, but some providers such as [ProfitBricks](https://www.profitbricks.com/) or [ElasticHosts](https://www.elastichosts.com/) do not have the hardware profiles concept and the previous implementation provides a fixed configuration with a fixed (hardcoded) list just to conform the interface. The new implementation allows to use custom hardwares or both (when supported) hardware profiles and custom hardwares.
+
+### Providers supported
+There are several providers that support arbitrary values of CPU and RAM like Docker, ElasticHosts, Google Compute Engine, etc. The first available providers supported by the new feature are:
+
+* [Google Compute Engine](https://cloud.google.com/compute/)
+* [ProfitBricks](https://www.profitbricks.com/)
+
+To configure the new feature in other providers add a bind() to the ArbitraryCpuRamTemplateBuilderImpl class at the provider's context module:
+
+{% highlight Java %}
+bind(TemplateBuilderImpl.class).to(ArbitraryCpuRamTemplateBuilderImpl.class);
+{% endhighlight %}
+
+Also is necessary to modify the function that transform a node from the provider model to the portable model of jclouds, to include the new automatic hardwareId (if apply).
+
+### How to create custom hardwares
```

Comment:
Don't say "create custom hardwares", as we're not creating anything. Better refer to the feature as "using custom hardwares".